thinc/tests/legacy/test_loss.py

```diff
@@ -0,0 +1,252 @@
+import pytest
```

Comment:
I would prefer to keep all the tests in the same file, that'll also help with tracking the changes in this PR. So basically where we're testing the old behaviour, we can rename the function to something with `legacy` in it and retrieve the correct class from the registry instead of importing directly. The rest should then remain the same, which will clearly showcase that the functionality can be non-breaking if you ajdust the script slightly to fetch the older version from registry.

Reply:
And then we'd have entirely new testing methods for the new functionality.